packages/material-ui/src/ClickAwayListener/ClickAwayListener.js

```diff
@@ -123,32 +123,23 @@ const ClickAwayListener = React.forwardRef(function ClickAwayListener(props, ref
   }, [handleClickAway, mouseEvent]);
 
   // Keep track of mouse/touch events that bubbled up through the portal.
-  const handleSyntheticMouse = (event) => {
+  const handleSynthetic = (handlerName) => (event) => {
     syntheticEventRef.current = true;
 
     const childrenProps = children.props;
-    if (childrenProps[mouseEvent]) {
-      childrenProps[mouseEvent](event);
-    }
-  };
-
-  const handleSyntheticTouch = (event) => {
-    syntheticEventRef.current = true;
-
-    const childrenProps = children.props;
-    if (childrenProps[touchEvent]) {
-      childrenProps[touchEvent](event);
+    if (childrenProps[handlerName]) {
+      childrenProps[handlerName](event);
     }
   };
 
   const childrenProps = {};
 
   if (mouseEvent !== false) {
-    childrenProps[mouseEvent] = handleSyntheticMouse;
+    childrenProps[mouseEvent] = handleSynthetic(mouseEvent);
   }
 
   if (touchEvent !== false) {
-    childrenProps[touchEvent] = handleSyntheticTouch;
+    childrenProps[touchEvent] = handleSynthetic(touchEvent);
   }
 
   return (
```

Comment:
another small optimisation to avoid object spread is to do 

```js
const childrenProps = { ref: handleRef };
```

Reply:
```diff
-  var childrenProps = {};
+  var childrenProps = {
+    ref: handleRef
+  };
 
   if (mouseEvent !== false) {
     childrenProps[mouseEvent] = handleSyntheticMouse;
@@ -133,9 +134,7 @@ var ClickAwayListener = React.forwardRef(function ClickAwayListener(props, ref)
     childrenProps[touchEvent] = handleSyntheticTouch;
   }
 
-  return /*#__PURE__*/React.createElement(React.Fragment, null, React.cloneElement(children, _extends({
-    ref: handleRef
-  }, childrenProps)));
+  return /*#__PURE__*/React.createElement(React.Fragment, null, React.cloneElement(children, childrenProps));
```

Reply:
This sounds like a great idea

Reply:
In a different PR maybe we could consider using a Babel plugin to optimise away `Fragment` with 1 child. I know you're currently doing it so that `react-docgen` can recognise components, but it's kinda dead code.

```jsx
- return <React.Fragment>{React.cloneElement(children, childrenProps)}</React.Fragment>;
- return React.cloneElement(children, childrenProps)};
```